js/webtorrent/entry.js

```diff
@@ -1,148 +1,186 @@
-/* global Blob, URL */
-
 const ipc = window.chrome.ipcRenderer
-const messages = require('../constants/messages')
+
+const clipboardCopy = require('clipboard-copy')
 const parseTorrent = require('parse-torrent')
+const path = require('path')
+const querystring = require('querystring')
 const React = require('react')
 const ReactDOM = require('react-dom')
+const url = require('url')
 const WebTorrentRemoteClient = require('webtorrent-remote/client')
 
-// React Components
-const MediaViewer = require('./components/mediaViewer')
-const TorrentViewer = require('./components/torrentViewer')
+const App = require('./components/app')
+const messages = require('../constants/messages')
 
 // Stylesheets
 require('../../less/webtorrent.less')
 require('../../node_modules/font-awesome/css/font-awesome.css')
 
-// UI state object. Pure function from `state` -> React element.
-const state = {
-  torrentID: window.decodeURIComponent(window.location.hash.substring(1)),
-  parsedTorrent: null,
-  client: null,
+// Pure function from state -> React elements.
+const store = {
+  name: null, // Torrent name
+  ix: null, // Selected file index
+  torrentId: window.decodeURIComponent(window.location.hash.substring(1)),
+  torrentIdProtocol: null,
   torrent: null,
+  serverUrl: null,
   errorMessage: null
 }
-window.state = state /* for easier debugging */
 
-state.parsedTorrent = parseTorrent(state.torrentID)
+let client, server
+
+init()
+
+function init () {
+  const parsedUrl = url.parse(store.torrentId)
+  store.torrentIdProtocol = parsedUrl.protocol
+
+  // `ix` param can be set by query param or hash, e.g. ?ix=1 or #ix=1
+  if (parsedUrl.protocol === 'http:' || parsedUrl.protocol === 'https:') {
```

Comment:
Fix race condition where media is briefly added to <iframe>  …
If there is no file object yet, but there is a serverUrl, then an
iframe is added to the page, instead of a "Loading..." message